chain/signatures.go

```diff
@@ -0,0 +1,53 @@
+package chain
```

Comment:
Not a fan of putting things in the chain package, can easily generate dependency loops

Reply:
Ack, if that's the case we can move out. This used to be in a signatures package, but @arajasek asked to move it out. We can move it somewhere else easily.